### DISCUSSION TITLE
Both client and controller on same device?

### DISCUSSION THREAD

**Discussion by glblduh** [dec0dOS/zero-ui#61] [Q&A] [answered]
Can this be used with a zerotier client on the host and the controller in a docker controller?

**Comment by dec0dOS** [comment]
Yes, the client and controller could be used on the same device. I'd recommend you to use the ZeroTier on the host, outside the docker in that case. For more info check the following thread: https://github.com/dec0dOS/zero-ui/discussions/23

Or you could use two ZeroTier instances: one for the controller in the Docker, and the second on the host as a client.

**Reply by glblduh** [reply]
Wouldn't the exposed port 9993 of the container overlap with the port 9993 of the zerotier instance in the host?

**Reply by dec0dOS** [reply]
No, it won't overlap. ZeroTier is smart enough to use different port. Just don't block the 9994 UDP port in the firewall.

**Reply by glblduh** [reply]
Do I need to change the mapped port of ZeroTier in the compose file?

**Comment by glblduh** [comment, marked as answer]
I've got it to work by changing the mapped host port of the container to 9994.

**Reply by dec0dOS** [reply]
Please not that you should not expose the 9994/tcp port to WAN, only the 9994/udp.

**Reply by glblduh** [reply]
Yep, I only exposed the UDP port.
